.github/workflows/back-compat.yml

```diff
@@ -0,0 +1,41 @@
+name: back-compat
+
+on:
+  push:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+env:
+  GO_VERSION: '1.22.5'
+
+jobs:
+  check-back-compat:
+    name: build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 2
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: ${{ env.GO_VERSION }}
+    - name: build head
+      run: |
+        go build -o omm_head
+        cp omm_head /var/tmp
+        rm omm_head
+    - run: git checkout HEAD~1
+    - name: build main
+      run: |
+        go build -o omm_prev
+        cp omm_prev /var/tmp
+    - name: Run last version
+      run: |
+        /var/tmp/omm_prev --db-path=omm-test.db 'test: a task from previous commit'
```

Comment:
Are you sure you will be able to merge/release if a bug occur ?

Maybe because a library changed, or you bumped go version.

 I mean if code is KO, this will fail, so you won't be able to merge?

Reply:
I didn't fully understand. Could you describe the scenario a bit more?

Reply:
You are adding something to the workflow.

So to something that may prevent you to do errors

Now imagine you merged something, and because of something outside your code. It could be the go version, an external package, a tool… your tests no longer pass, even if they did before.

So now your tests are failing.
So you provide a fix
You push
The workflow is run
The code that check if it's a regression, the code you added with the PR about the workflow, will fail because tests are failing.

So now the workflow is failing, won't it prevent you to merge the fix ?